Gems/AtomLyIntegration/CommonFeatures/Code/Source/CoreLights/DiskLightDelegate.cpp

```diff
@@ -47,62 +49,91 @@ namespace AZ::Render
         return m_shapeBus->GetRadius() * GetTransform().GetUniformScale();
     }
 
-    void DiskLightDelegate::DrawDebugDisplay(const Transform& transform, [[maybe_unused]]const Color&, AzFramework::DebugDisplayRequests& debugDisplay, bool isSelected) const
+    DiskLightDelegate::DiskVisualizationDimensions DiskLightDelegate::CalculateDiskVisualizationDimensions(const float degrees) const
     {
-        debugDisplay.PushMatrix(transform);
         const float radius = GetConfig()->m_attenuationRadius;
         const float shapeRadius = m_shapeBus->GetRadius();
 
-        auto DrawConicalFrustum = [&debugDisplay](uint32_t numRadiusLines, const Color& color, float brightness, float topRadius, float bottomRadius, float height)
+        const float radians = DegToRad(degrees);
+        const float coneRadius = Sin(radians) * radius;
+        const float coneHeight = Cos(radians) * radius;
+
+        return DiskVisualizationDimensions{ shapeRadius, shapeRadius + coneRadius, coneHeight };
+    }
+
+    Aabb DiskLightDelegate::GetLocalVisualizationBounds() const
+    {
+        const auto [radius, height] = [this]
+        {
+            if (GetConfig()->m_enableShutters)
+            {
+                const auto [innerTopRadius, innerBottomRadius, innerHeight] = CalculateDiskVisualizationDimensions(
+                    GetMin(GetConfig()->m_outerShutterAngleDegrees, GetConfig()->m_innerShutterAngleDegrees));
+                const auto [outerTopRadius, outerBottomRadius, outerHeight] =
+                    CalculateDiskVisualizationDimensions(GetConfig()->m_outerShutterAngleDegrees);
+                return AZStd::pair{ GetMax(outerBottomRadius, GetMax(innerBottomRadius, GetMax(innerTopRadius, outerTopRadius))),
+                                    GetMax(innerHeight, outerHeight) };
+            }
+
+            const auto [topRadius, bottomRadius, height] = CalculateDiskVisualizationDimensions(DefaultConeAngleDegrees);
+            return AZStd::pair{ GetMax(topRadius, bottomRadius), height };
+        }();
+
+        return Aabb::CreateFromMinMax(Vector3(-radius, -radius, 0.0f), Vector3(radius, radius, height));
+    }
```

Comment:
I can't help but feel this is doing a lot more work than necessary, which also makes it hard to read what's actually happening. We really only need to do 3 things to make the aabb:
1. Cap the inner angle be no greater than the outer angle.
2. Get the radius from the outer angle. (The outer angle will always have the greater radius)
3. Get the height from the inner angle. (The inner angle will always have the greater height)